File: src/test/java/workshop/part2/ComputeTestActor.java
package workshop.part2;

import akka.actor.AbstractActor;
import akka.japi.pf.ReceiveBuilder;
import scala.PartialFunction;
import scala.runtime.BoxedUnit;

public class ComputeTestActor extends AbstractActor {
    Boolean restarted = false;

    @Override
    public void postRestart(Throwable reason) throws Exception {
        restarted = true;
    }

    @Override
    public PartialFunction<Object, BoxedUnit> receive() {
        return ReceiveBuilder.
                match(Exception.class, e -> {
                    throw e;
                }).
                match(IsRestarted.class, m -> sender().tell(restarted, self())).build();
    }

    static class IsRestarted {}
}

File: src/test/scala/workshop/part2/ComputeTestActor.scala
package workshop.helpers

import akka.actor.Actor
import akka.event.Logging


case class IsRestarted()

class ComputeTestActor extends Actor {
  val log = Logging(context.system, this)
  var restarted = false

  override def postRestart(reason: Throwable): Unit = {
    restarted = true
  }

  override def receive = {
    case e: Exception => {
      throw e
    }
    case IsRestarted => sender ! restarted
  }
}